### ISSUE TITLE
Support miter limits on stroke joins

### ISSUE THREAD

**Issue by rlafuente** [shoebot/shoebot#418]
Quick one, just a wrapper around cairo's [set_miter_limit](https://www.cairographics.org/manual/cairo-cairo-t.html#cairo-set-miter-limit)

Possible syntax: `strokejoin(MITER, limit=0.5)`

**Comment by stuaxo** [comment]
OK, this makes sense that you can add parameters relevant to whatever kind of join you are using to strokejoin.

Does/should stroke(...) accept analgous parameters, e.g.: `stroke(join=MITER, miter_limit=0.5)` ?

**Comment by rlafuente** [comment]
Yeah it's really a super minor effect, but i ran into a small glitch caused by the miterlimit when animating paths today, and remembered that i put off implementing it during the fill/stroke work because it was too low priority. It properly bit back :woman_facepalming: 

And yes definitely! While i'm with you with underscores for argument/function names being more Python proper, the Nodebox standard would probably be `miterlimit` (i followed that for other stroke commands like strokedash)
